n3fit/src/n3fit/model_trainer.py

```diff
@@ -523,12 +518,43 @@ def _generate_observables(
         self._reset_observables()
         log.info("Generating layers")
 
+        # We need to transpose Experimental data, stacking over replicas
+        experiment_data = {
+            "trmask": [],
+            "expdata": [],
+            "expdata_vl": [],
+            "invcovmat": [],
+            "invcovmat_vl": [],
+        }
+
+        # Loop over datasets
+        for i in range(len(self.exp_info[0])):
+            # Loop over data fields
+            for key, value in experiment_data.items():
+                replica_data = []
+                # Loop over replicas
+                for replica in self.exp_info:
+                    if key in ["expdata", "expdata_vl"]:
+                        replica_data.append(replica[i][key].flatten())
+                    else:
+                        replica_data.append(replica[i][key])
```

Comment:
```suggestion
                    replica_data.append(replica[i][key].flatten())
```

Better to flatten the 1-d array as well

Reply:
I just checked, this doesn't work (shape mismatch of 5776(=76*76) vs 76 in DEUTERON), so it's not a 1d array.

Reply:
Uh, why is the training data ndata x ndata ? That doesn't look right.

Reply:
I just checked and all the quantities are (1, ndata)

Reply:
ah, all the quantities that are data ofc, let me change the suggestion to a comment then